ssh/tailssh/tailssh.go

```diff
@@ -244,19 +245,37 @@ ProcessAction:
 		return
 	}
 
-	var ctx context.Context = context.Background()
-	if action.SesssionDuration != 0 {
-		sctx := newSSHContext()
-		ctx = sctx
-		t := time.AfterFunc(action.SesssionDuration, func() {
-			sctx.CloseWithError(userVisibleError{
-				fmt.Sprintf("Session timeout of %v elapsed.", action.SesssionDuration),
-				context.DeadlineExceeded,
-			})
-		})
-		defer t.Stop()
+	ss := &sshSession{
+		Session:   s,
+		idH:       s.Context().(ssh.Context).SessionID(),
+		ctx:       newSSHContext(),
+		srv:       srv,
+		action:    action,
+		localUser: lu,
+		connInfo:  ci,
+		logf:      srv.logf,
 	}
-	srv.handleAcceptedSSH(ctx, s, ci, lu)
+	ss.run()
+}
+
+// sshSession is an accepted Tailscale SSH session.
+type sshSession struct {
+	ssh.Session
+	idH      string // the RFC4253 sec8 hash H; don't share outside process
+	sharedID int64  // ID that's shared with control; time.Now().UnixNano()
```

Comment:
why not just use a random ID here instead of UnixNano?

Reply:
I was thinking of sorting and lifetime uniqueness.

But I can do date+randhex.